### PR TITLE
[@testing-library/dom] update to v7

### PR DESCRIPTION
**Pull request by abraham**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/testing-library/dom-testing-library/releases/tag/v7.0.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by kentcdodds — APPROVED** [review]
Lovely! Thank you 👍

**Review by eps1lon — CHANGES_REQUESTED** [review]

File: types/testing-library__dom/wait-for-element-to-be-removed.d.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export function waitForElementToBeRemoved<T>(
-    callback: () => T,
+    callback: () => T | HTMLElement,
```
[the remark this answers is not in the document]
Comment (abraham) [reply types/testing-library__dom/wait-for-element-to-be-removed.d.ts]: Updated

File: types/testing-library__dom/wait-for-element-to-be-removed.d.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 export function waitForElementToBeRemoved<T>(
-    callback: () => T,
+    callback: (() => T) | HTMLElement | HTMLElement[],
```
Comment (eps1lon) [inline types/testing-library__dom/wait-for-element-to-be-removed.d.ts]:
```suggestion
    callback: (() => T) | T,
```
is how we document this as well.

**Review by eps1lon — APPROVED** [review]
Perfect, thanks!

**Review by kentcdodds — APPROVED** [review]
💯💯

**Review by timdeschryver — COMMENTED** [review]
Should we also add deprecated comments to `wait`, `waitForDomChange` and `waitForElement`?

**Comment by kentcdodds** [comment]
Those are added here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43108

**Comment by amaechler** [comment]
Should the [`pretty-format`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ff99f750027135848647b20da5e0e4ee59d486e3/types/testing-library__dom/package.json#L4) dependency be updated to `^25.1.0` as well to [match](https://github.com/testing-library/dom-testing-library/blob/master/package.json#L47) the library?

**Comment by kentcdodds** [comment]
Yes it should

**Comment by abraham** [comment]
Updated `pretty-format`

**Comment by amaechler** [comment]
Awesome, thanks.

**Comment by jrwpatterson** [comment]
when will this be merged?

**Comment by eps1lon** [comment]
> when will this be merged?

Microsoft employees handle merging. And as you might expect considering the global situation things might go slower than usual. You'll get this eventually.

In the meantime you can always patch this locally:

```ts
declare module '@testing-library/dom' {
    export function waitFor(
        callback: () => void,
        options?: {
            container?: HTMLElement;
            timeout?: number;
            interval?: number;
            mutationObserverOptions?: MutationObserverInit;
        },
    ): Promise<void>;
}
```

**Comment by bopfer** [comment]
The type in this PR  for `waitFor` indicates that it aways returns `void`.  But, in the [release notes](https://github.com/testing-library/dom-testing-library/releases/tag/v7.0.0) it says that it can also be used to replace `waitForElement`, which would return an `HTMLElement`.  Should the types be updated to reflect that?

**Comment by wKovacs64** [comment]
@bopfer Sure seems that way. Like this, I think?

```ts
export function waitFor<T>(
    callback: () => T,
    options?: {
        container?: HTMLElement;
        timeout?: number;
        interval?: number;
        mutationObserverOptions?: MutationObserverInit;
    },
): Promise<T>;
```

**Comment by abraham** [comment]
@wKovacs64 updated

**Comment by orta** [comment]
This CI fail is on us, not on this PR, I think this is fine to merge 👍

**Comment by abraham** [comment]
Thanks @orta

**Comment by ClementeSerrano** [comment]
Hi guys! I still having the TS error:

`@types/testing-library__react"' has no exported member 'waitFor'.`

I'm on version `10.0.1` of the package

Any updates?

**Comment by abraham** [comment]
@ClementeSerrano The react update is over in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43108

**Comment by ClementeSerrano** [comment]
Cool. Thanks @abraham , I'll be attentive to the merge ;)

**Pull request merged**